README.md

```diff
@@ -19,9 +35,181 @@ Install Docker and look at the Docker instructions.
+  ```
+  kubectl set image deployment/upgrade-analyzer-deployment \
+   upgrade-analyzer=gcr.io/<google-project>/ci:<build-number>
+   ^^ deploy will not occur if you don't set a new image build-number
+      if you want to force a pods recreation with same ci:<buildnumber>
+      simply delete the running pods with 'kubectl delete pods POD_NAME'
+      the replication controller will automatically recreate a new one
+   ```
+
+### Previous (blunt) way to deploy imperatively ( reference only )
+  ```
+  kubectl delete deployment upgrade-analyzer-deployment
+  kubectl run upgrade-analyzer-deployment \
+    --image=gcr.io/<google-project>/ci:<build-number> --command \
+    -- /opt/ci/bin/upgrade_analyzer --listen --repo=REPO --token=GITHUB_TOKEN
+  ```
+
+### Useful kubernetes commands
+  ```
+  kubectl get pods
+  kubectl get deployments
+  kubectl get secrets
+  kubectl logs PODS_NAME
+  kubectl exec -it PODS_NAME -- sh
+  kubectl run -i --tty test --image=gcr.io/<google-project>/ci:<build-number> -- sh
+  kubectl delete deployment NAME_OF_DEPLOYMENT
+  kubectl describe secret/upgrade-analyzer-secrets
+  kubectl get secret upgrade-analyzer-secrets -o yaml  # secrets show as base64
+  kubectl config view
+  kubectl proxy
+  kubectl cluster-info # see URI for kubernetes-dashboard
+  ^^  See your google cloud console for the credential of the GKE cluster
+  ```
+
+### External Documentation
 
-### Docker
+* https://help.github.com/articles/creating-an-access-token-for-command-line-use/
+* https://cloud.google.com/container-engine/docs/quickstart
+* https://github.com/travis-ci/travis.rb#table-of-contents
+* http://octokit.github.io/octokit.rb/Octokit.html
+* http://kubernetes.io/docs/user-guide/secrets/
+* http://kubernetes.io/docs/
+* https://developers.google.com/console/help/new/#serviceaccounts
+* https://cloud.google.com/container-registry/docs/advanced-authentication
+* https://docs.travis-ci.com/user/customizing-the-build
+* https://docs.travis-ci.com/user/deployment/script/
+* https://deis.com/blog/2016/kubernetes-illustrated-guide/?imm_mid=0e520d&cmp=em-prog-na-na-newsltr_20160625
 
-    host$ docker build -t upgrade_toolbelt .
-    host$ docker run --rm --env 'UTB_OWNER_AND_REPO=benjaminoakes/maid' --env 'UTB_JOB_COUNT=11' -v "$PWD:/opt" -v "$HOME/var/ci":/var/ci -i -t upgrade_toolbelt sh -l
-    container$ utb-build-analyze 407
+### Alternate way to authenticate to GKE/GCR
+( for CI integration purpose where gcloud sdk is not available )
+  ```
+  docker login -u _json_key -p="${GCE_JSON_KEY}" https://gcr.io
+  docker push gcr.io/${GCE_PROJECT}/ci:${TRAVIS_BUILD_NUMBER}
+  kubectl config set-credentials default --username=${GKE_USERNAME} --password=${GKE_PASSWORD}
+  kubectl config set-cluster default --server=http://${GKE_SERVER} --insecure-skip-tls-verify=true
+  kubectl config set-context default --cluster=default --user=default
+  kubectl config use-context default
+  ```
+## Docker for the upgrade_toolbelt tool ( legacy )
+  ``
```

Comment:
ended up pasting the whole file in http://dillinger.io/
hopefully didn't miss anything else.